src/Models/HydrostaticFreeSurfaceModels/fft_based_implicit_free_surface_solver.jl

```diff
@@ -110,8 +110,8 @@ end
 
 @kernel function fft_implicit_free_surface_right_hand_side!(rhs, grid, g, Lz, Δt, ∫ᶻQ, η)
     i, j = @index(Global, NTuple)
-    Az = Azᶜᶜᶜ(i, j, 1, grid)
+    Az = Azᶜᶜᶜ(i, j, grid.Nz, grid)
     δ_Q = flux_div_xyᶜᶜᶜ(i, j, 1, grid, ∫ᶻQ.u, ∫ᶻQ.v)
-    @inbounds rhs[i, j, 1] = (δ_Q - Az * η[i, j, 1] / Δt) / (g * Lz * Δt * Az)
+    @inbounds rhs[i, j, grid.Nz] = (δ_Q - Az * η[i, j, grid.Nz] / Δt) / (g * Lz * Δt * Az)
```

Comment:
`rhs` is a 2D array I believe so perhaps this should be
```suggestion
    @inbounds rhs[i, j, 1] = (δ_Q - Az * η[i, j, grid.Nz] / Δt) / (g * Lz * Δt * Az)
```
?